### PR TITLE
ZIP 316: Transparent ovk

### PR DESCRIPTION
**Pull request by therealyingtong**
This is based on the https://github.com/zcash/zcash/tree/feature/wallet_unified_addresses branch.

This PR makes use of https://github.com/zcash/librustzcash/pull/341.

### CODE REVIEW

File: src/rust/src/unified_keys_ffi.rs [hunk 1]
```diff
@@ -129,3 +132,41 @@ pub extern "C" fn unified_full_viewing_key_from_components(
...
+                *out = fvk.external_ovk().as_bytes();
+            } else {
+                *out = fvk.internal_ovk().as_bytes();
+            }
+
+            return true;
+        }
+    }
+
+    false
```
Comment (daira) [inline src/rust/src/unified_keys_ffi.rs:171]: Use `find_ovk` defined at https://github.com/zcash/librustzcash/pull/480/files#diff-3b307a8fabc1b1ac923d7b23188c2fd995a416495c210a148a9914e7728d503aR86

```suggestion
    if let Some(ovk) = find_ovk(key, external) {
        *out = ovk.as_bytes();
        true
    } else {
        *out = [0; 32];
        false
    }
```
(I'm not sure if `*out = [0; 32];` is correct. In any case, zeroing the output array on error is probably a good idea.)

**Review by daira — APPROVED** [review]
utACK modulo comment.

File: src/rust/src/unified_keys_ffi.rs (hunk 1 above)
Comment (therealyingtong) [reply src/rust/src/unified_keys_ffi.rs]: Resolving because the implementation has changed significantly.

**Comment by therealyingtong** [comment]
Closing in favour of https://github.com/zcash/zcash/pull/5469/commits/fe005c76f8ac3905a7da55715be00bfb1dc11bf3.

**Pull request closed without merging**